shared/storage.ts

```diff
@@ -312,34 +315,62 @@ class Storage {
     }
   }
 
+  private static async verifyCredential(
+    verifiableCredential: Credential,
+    issuer: string,
+  ) {
+    let isVerified = true;
+    if (isMosipVC(issuer)) {
+      const verificationResult = await verifyCredential(verifiableCredential);
+      isVerified = verificationResult.isVerified;
+    }
+    return isVerified;
+  }
+
   private static async loadVCs(completeBackupData: {}, encryptionKey: any) {
     try {
       const allVCs = completeBackupData['VC_Records'];
       const allVCKeys = Object.keys(allVCs);
       const dataFromDB = completeBackupData['dataFromDB'];
+
       // 0. Check for VC presense in the store
       // 1. store the VCs and the HMAC
-      allVCKeys.forEach(async key => {
-        let vc = allVCs[key];
-        const ts = Date.now();
-        const prevUnixTimeStamp = vc.vcMetadata.timestamp;
-        vc.vcMetadata.timestamp = ts;
-        dataFromDB.myVCs.forEach(myVcMetadata => {
-          if (
-            myVcMetadata.requestId === vc.vcMetadata.requestId &&
-            myVcMetadata.timestamp === prevUnixTimeStamp
-          ) {
-            myVcMetadata.timestamp = ts;
-          }
-        });
-        const updatedVcKey = new VCMetadata(vc.vcMetadata).getVcKey();
-        const encryptedVC = await encryptJson(
-          encryptionKey,
-          JSON.stringify(vc),
-        );
-        // Save the VC to disk
-        await this.setItem(updatedVcKey, encryptedVC, encryptionKey);
-      });
+      // 2. Verify the VC and update the result
+
+      await Promise.all(
+        allVCKeys.map(async key => {
```

Comment:
Why are we using map here. We are not returning any object. This processing can be achieved by forEach.

Reply:
The difference between `promise.all()` + `array.map()` and `.forEach()` is that it's impossible to await a `forEach()` itself. We can run tasks in parallel in `.forEach()`  just like you can with `.map()`, but we can't wait for all of those parallel tasks to finish, then do something once they've all finished. The whole point of using `.map()` instead of `.forEach()`  is so you can get a list of promises, collect them with `Promise.all()`, then await the whole thing. 

Since we are calling the `verifyCredential()` which is a async method, we have to use `.map()` with `Promise.all()`